k9mail/src/main/java/com/fsck/k9/provider/AttachmentTempFileProvider.java

```diff
@@ -95,7 +96,7 @@ public static Uri getMimeTypeUri(Uri contentUri, String mimeType) {
     public static boolean deleteOldTemporaryFiles(Context context) {
         File tempDirectory = getTempFileDirectory(context);
         boolean allFilesDeleted = true;
-        long deletionThreshold = new Date().getTime() - FILE_DELETE_THRESHOLD_MILLISECONDS;
+        long deletionThreshold = SystemClock.elapsedRealtime() - FILE_DELETE_THRESHOLD_MILLISECONDS;
```

Comment:
Also wrong - we're providing a threshold date-time here, not a duration - so currentTimeMillis is what we want.